worlds/yoshisisland/level_logic.py

```diff
@@ -0,0 +1,486 @@
+from BaseClasses import CollectionState
+from typing import Optional, TYPE_CHECKING
+if TYPE_CHECKING:
+    from . import YIWorld
+
+class YoshiLogic:
+    player: int
+    game_logic: str
+    midring_start: bool
+    clouds_always_visible: bool
+    consumable_logic: bool
+    luigi_pieces: int
+
+
+    def __init__(self, world: Optional["YIWorld"]) -> None:
+        self.player = world.player
+        self.boss_order = world.boss_order
+        self.luigi_pieces = world.luigi_pieces
+
+        if world:
+            if world.options.stage_logic == 0:
+                self.game_logic = "Easy"
+            elif world.options.stage_logic == 1:
+                self.game_logic = "Normal"
+            else:
+                self.game_logic = "Hard"
+
+        if world:
+            self.midring_start = world.options.shuffle_midrings == 0
+            self.consumable_logic = world.options.item_logic != 0
+
+        if world:
+            if world.options.hidden_object_visibility >= 2:
+                self.clouds_always_visible = True
+            else:
+                self.clouds_always_visible = False
+
+        if world:
+            self.bowser_door = world.options.bowser_door_mode
+            if self.bowser_door not in {0, 1, 2, 5}:
+                self.bowser_door = 3

```

Comment:
```suggestion
            self.midring_start = world.options.shuffle_midrings == 0
            self.consumable_logic = world.options.item_logic != 0

            if world.options.hidden_object_visibility >= 2:
                self.clouds_always_visible = True
            else:
                self.clouds_always_visible = False

            self.bowser_door = world.options.bowser_door_mode
            if self.bowser_door not in {0, 1, 2, 5}:
                self.bowser_door = 3
```
No reason to check world 4 times when once is sufficient.

Reply:
done